test/people-route-test.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const People = require('../model/people.js');
+const PORT = process.env.PORT || 3000;
+const debug = require('debug')('people:people-route-test');
+
+process.env.MONGODB_URI = 'mongodb://localhost/peopletest';
+
+require('../server.js');
+
+const url = `http://localhost:${PORT}`;
+const examplePeople = {
+  name: 'test people name'
+};
+
+describe('People routes', function() {
+  describe('POST: /api/people', function() {
+    describe('with a valid body', function() {
+      after( done => {
+        if(this.temppeople) {
+          People.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+
+      it('should return a valid people', done => {
+        request.post(`${url}/api/people`)
+        .send(examplePeople)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          this.temppeople = res.body;
+          expect(res.body.name).to.equal('test people name');
+          done();
+        });
+      });
+    });
+
+    describe('with no body supplied', function() {
+      it('should return a 400 error', function() {
```

Comment:
you need to pass in done here not an anonymous function. If you look at your tests you'll notice that a few of the response seem to come out of order and this is happening because you don't pass in done so your tests because they are running async are returning in random order.